sklearn/cluster/_agglomerative.py

```diff
@@ -34,7 +34,7 @@
 )
 from ..utils.graph import _fix_connected_components
 from ..utils.validation import check_memory, validate_data
-from . import _hierarchical_fast as _hierarchical
+from . import _hierarchical_fast as _hierarchical  # type: ignore
```

Comment:
Once https://github.com/scikit-learn/scikit-learn/pull/30709 is merged I guess this is not needed anymore?